### ISSUE TITLE
Unable to load OData adapter from assembly Simple.OData.Client.V3.Adapter ---> System.InvalidOperationException: Sequence contains no matching element

### ISSUE THREAD

**Issue by russcam** [object/Simple.OData.Client#417]
Nuget package version: 5.0.0-alpha3

Attempting to use Simple.OData.Client (or the V3 package) to pull packages down from Nuget for an  example repo for getting started with Elasticsearch.

[I've ported it over to .net core 1.1](https://github.com/elastic/elasticsearch-net-example/tree/5.x-codecomplete-netcore) and all works on Windows 10, however, attempting to run the Harvester project on macOS with

```
~/src/elasticsearch-net-example/src/NuSearch.Harvester$ dotnet run -- ~/nuget-data
```

yields the following

```
Unhandled Exception: System.AggregateException: One or more errors occurred. (Unable to load OData adapter from assembly Simple.OData.Client.V3.Adapter) ---> System.InvalidOperationException: Unable to load OData adapter from assembly Simple.OData.Client.V3.Adapter ---> System.InvalidOperationException: Sequence contains no matching element
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source, Func`2 predicate)
   at Simple.OData.Client.AdapterFactory.LoadAdapter(String adapterAssemblyName, String adapterTypeName, Object[] ctorParams)
   --- End of inner exception stack trace ---
   at Simple.OData.Client.AdapterFactory.LoadAdapter(String adapterAssemblyName, String adapterTypeName, Object[] ctorParams)
   at Simple.OData.Client.AdapterFactory.<CreateAdapterAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Simple.OData.Client.Session.<ResolveAdapterAsync>d__21.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Simple.OData.Client.ODataClient.<FindScalarAsync>d__99.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Simple.OData.Client.BoundClient`1.<FindScalarAsync>d__11`1.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at NuSearch.Harvester.Program.Main(String[] args) in /Users/russ/src/elasticsearch-net-example/src/NuSearch.Harvester/Program.cs:line 24
```
It looks like it's able to load the assembly correctly, but [failing to find the intended ctor](https://github.com/object/Simple.OData.Client/blob/51984e23b00cf2e33a1b81eb279df8f3ad1908bc/Simple.OData.Client.Core/Adapter/AdapterFactory.cs#L120-L122).

dotnet --info

```
.NET Command Line Tools (1.0.4)

Product Information:
 Version:            1.0.4
 Commit SHA-1 hash:  af1e6684fd

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.12
 OS Platform: Darwin
 RID:         osx.10.12-x64
 Base Path:   /usr/local/share/dotnet/sdk/1.0.4
```
Any ideas that I can try?

**Comment by sixlettervariables** [comment]
I ran into this working on separate code to reduce metadata accesses and parsing, and it appears that the code needs to use `IsAssignableFrom` when checking for the correct ctor. See my updates here: https://github.com/sixlettervariables/Simple.OData.Client/blob/5d73f6ebccd753a9c3c2bac356229969148809e9/Simple.OData.Client.Core/Adapter/AdapterFactory.cs#L149-L157

**Comment by object** [comment]
Should be fixed by @sixlettervariables's PR. Released in 4.28.0.

**Comment by object** [comment]
And I also pushed version 5 alpha 4 that is incorporated changed from @sixlettervariables.
